cmake/DAGMCConfig.cmake.in

```diff
@@ -0,0 +1,22 @@
+# - Config file for the Direct Accelerated Geometry Monte Carlo (DAGMC) package
+# It defines the following variables
+#  DAGMC_INCLUDE_DIRS - include directories for FooBar
+#  DAGMC_LIBRARIES    - libraries to link against
+
+set(DAGMC_FOUND 1)
+
+set(DAGMC_MAJOR_VERSION @DAGMC_MAJOR_VERSION@)
+set(DAGMC_MINOR_VERSION @DAGMC_MINOR_VERSION@)
+set(DAGMC_PATCH_VERSION @DAGMC_PATCH_VERSION@)
+set(DAGMC_VERSION @DAGMC_VERSION@)
+  
+message(STATUS "Configuring DAGMC ${DAGMC_VERSION}")
+
+
+set(DAGMC_INCLUDE_DIRS "@CMAKE_INSTALL_PREFIX@/@INSTALL_INCLUDE_DIR@" "@MOAB_INCLUDE_DIRS@")
+set(DAGMC_LIBRARY_DIRS "@CMAKE_INSTALL_PREFIX@/@INSTALL_LIB_DIR@")
+set(DAGMC_LINK_LIBS "-L@CMAKE_INSTALL_PREFIX@/@INSTALL_LIB_DIR@/ -ldagmc -ldagtally -lmakeWatertight -lpyne_dagmc" "@DAG_LINK_LIBS@")
```

Comment:
I think the -L part shouldn't be in `DAGMC_LINK_LIBS`. Also I think these would be better as lists, i.e.

`set(DAGMC_INCLUDE_DIRS @CMAKE_INSTALL_PREFIX@/@INSTALL_INCLUDE_DIR@ @MOAB_INCLUDE_DIRS@)`
`set(DAGMC_LIBRARY_DIRS @CMAKE_INSTALL_PREFIX@/@INSTALL_LIB_DIR@)`
`set(DAGMC_LINK_LIBS dagmc dagtally makeWatertight pyne_dagmc @DAG_LINK_LIBS@)`

And thinking about this even further, how often would anyone need to link against all of these at the same time?

Reply:
I'm a little confused. Why wouldn't we want the `-L` in that variable? My goal was to have two variables, `DAGMC_LINK_LIBS` which will provide all of the standard libraries built and installed with DAGMC and `DAGMC_LIBRARY_DIRS` so that applications only looking to link against one of those libraries could use something like:

`TARGET_LINK_LIBRARIES(<target> ${DAGMC_LIBRARIES} dagmc)`

I have less expertise with CMake than you do though @ljacobson64, so let me know if there are better practices.

Reply:
The best practice for link directories is to use `link_directories()`, which works just like `include_directories()`. So in our case, we'd use

```
target_link_libraries(<target> ${DAGMC_LIBRARIES})
link_directories(${DAGMC_LIBRARY_DIRS})
include_directories(${DAGMC_INCLUDE_DIRS})
```

Reply:
Gotcha. I'll adjust the variables accordingly then. Thanks! 

- [ ] Remove install lib dir from `DAGMC_LINK_LIBS` and rename to `DAGMC_LIBRARIES`
- [ ] Make new `DAGMC_LIBRARIES` variable a CMake list

Reply:
Ugh to there's a comma after the `MOAB_LIBRARY_DIRS` variable name in MOAB's CMake configuration file which causes it to be ignored when trying to use it in DAGMC. This makes it impossible to forward that library directory to other packages building on DAGMC. 

I've made an issue in their repo and will submit a PR, but I'm not sure what to do in the meantime other than to parse it out of another variable. Which is what I'll do for now. Not prety 😞